test/test_cmd_vel_mux.py

```diff
@@ -0,0 +1,223 @@
+
+
+def generate_test_description():
+    path_to_cmd_vel_mux_params = launch.substitutions.PathJoinSubstitution([
+        launch_ros.substitutions.FindPackageShare(package='cmd_vel_mux'),
+        'config',
+        'cmd_vel_mux_params.yaml'
+    ])
+    return launch.LaunchDescription([
+        launch_ros.actions.Node(
+            package='cmd_vel_mux',
+            node_executable='cmd_vel_mux_node',
+            parameters=[path_to_cmd_vel_mux_params],
+            output='screen'
+        ),
+        launch_testing.actions.ReadyToTest()
+    ])
+
+
+class TestCmdVelMux(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(
+        cls,
+        launch_service,
+        proc_info,
+        proc_output
+    ):
+        @contextlib.contextmanager
+        def launch_topic_command(self, arguments):
+            topic_command_action = launch.actions.ExecuteProcess(
+                cmd=['ros2', 'topic', *arguments],
+                additional_env={'PYTHONUNBUFFERED': '1'},
+                name='ros2topic-cli',
+                output='screen'
+            )
+            with launch_testing.tools.launch_process(
+                launch_service, topic_command_action, proc_info, proc_output
+            ) as topic_command:
+                yield topic_command
+        cls.launch_topic_command = launch_topic_command
+
+    def test_idle_mux(self):
+        with self.launch_topic_command(
+            arguments=[
+                'echo',
+                '--qos-reliability', 'reliable',
+                '--qos-durability', 'transient_local',
+                'active'
+            ]
+        ) as command:
+            assert command.wait_for_output(functools.partial(
+                launch_testing.tools.expect_output, expected_lines=[
+                    'data: idle',
+                    '---'
+                ], strict=True
+            ), timeout=2)
+        with self.launch_topic_command(arguments=['echo', 'cmd_vel']) as command:
+            assert not command.wait_for_output(lambda output: len(output) > 0, timeout=2)
+
+    def test_mux_with_single_input(self):
+        default_twist = geometry_msgs.msg.Twist()
+        default_twist.linear.x = 1.0
+        default_twist_block_yaml = message_to_yaml(default_twist)
+        default_twist_inline_yaml = yaml.dump(yaml.load(default_twist_block_yaml))
+        with self.launch_topic_command(
+            arguments=[
+                'pub', '-r', '20',
+                'input/default',
+                'geometry_msgs/msg/Twist',
+                f'{default_twist_inline_yaml}'
+            ]
+        ):
+            with self.launch_topic_command(
+                arguments=[
+                    'echo',
+                    '--qos-reliability', 'reliable',
+                    '--qos-durability', 'transient_local',
+                    'active'
+                ]
+            ) as command:
+                assert command.wait_for_output(functools.partial(
```

Comment:
Elsewhere I think we decided that using `self.assertTrue` is better in `unittest` tests, as it will continue the test and report all of the failures at the end.  Suggest using that here and everywhere else in this PR.